dace/transformation/interstate/state_fusion.py

```diff
@@ -207,6 +207,9 @@ def can_be_applied(self, graph, expr_index, sdfg, permissive=False):
             for e in in_edges:
                 if e.data.assignments.keys() & symbols_used:
                     return False
+                # Also fail in the inverse; symbols assigned on the second edge are free symbols on the first edge
+                if out_edges[0].data.assignments.keys() & set(e.data.free_symbols):
```

Comment:
you could use `new_assignments` here instead for performance

Reply:
Done